pkg/aws/cloud.go

```diff
@@ -38,6 +40,12 @@ type Cloud interface {
 	// creates lattice tags with default values populated and merges them with provided tags
 	DefaultTagsMergedWith(services.Tags) services.Tags
 
+	// find tags on lattice resources
+	FindTagsForARNs(ctx context.Context, arns []string) (map[string]services.Tags, error)
+
+	// find lattice target group ARNs using tags
+	FindTargetGroupARNs(context.Context, services.Tags) ([]string, error)
+
```

Comment:
I like the idea of including these methods in the Tagging() interface. I think we should change the current interface to remove the inherited ```taggingapiiface.ResourceGroupsTaggingAPIAPI``` interface. Then you can have two implementations, one that uses the tagging API and one that uses the Lattice tag APIs. Then, in ```NewCloud(...)``` you can check the ```TaggingServiceAPIDisabled``` option and assign one implementation or the other, rather than having switches inside each method.